### PR TITLE
Use `StateChildren`-based lookup for signature metadata

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
- Provides a `SigMetadataLookup` implementation backed by an injected `StateChildren`.
- Sets the foundation for expanding signatures from the latest signed state.

**Related issue(s)**:
- Addresses the main architectural pre-req for #2019

### CODE REVIEW

File: hedera-node/src/test/java/com/hedera/services/sigs/metadata/lookups/HfsSigMetaLookupTest.java [hunk 1]
```diff
@@ -1,114 +0,0 @@
-package com.hedera.services.sigs.metadata.lookups;
-
-/*-
- * ‌
- * Hedera Services Node
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/test/java/com/hedera/services/sigs/metadata/lookups/HfsSigMetaLookupTest.java:5]: This test class is deleted accidentally ?

File: hedera-node/src/main/java/com/hedera/services/sigs/metadata/TokenMetaUtils.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package com.hedera.services.sigs.metadata;
+
+import com.hedera.services.state.merkle.MerkleToken;
+import com.hedera.services.state.submerkle.FcCustomFee;
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/main/java/com/hedera/services/sigs/metadata/TokenMetaUtils.java:4]: copyright missing

File: hedera-node/src/main/java/com/hedera/services/sigs/metadata/StateChildrenSigMetadataLookup.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
+package com.hedera.services.sigs.metadata;
+
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/main/java/com/hedera/services/sigs/metadata/StateChildrenSigMetadataLookup.java:2]: copyright missing

File: hedera-node/src/test/java/com/hedera/services/sigs/metadata/StateChildrenSigMetadataLookupTest.java [hunk 4]
```diff
@@ -0,0 +1,384 @@
+package com.hedera.services.sigs.metadata;
+
+import com.hedera.services.config.MockFileNumbers;
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/test/java/com/hedera/services/sigs/metadata/StateChildrenSigMetadataLookupTest.java:3]: same here

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM !

File: hedera-node/src/main/java/com/hedera/services/sigs/metadata/StateChildrenSigMetadataLookup.java [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+import static com.hedera.services.sigs.order.KeyOrderingFailure.MISSING_SCHEDULE;
+import static com.hedera.services.sigs.order.KeyOrderingFailure.MISSING_TOKEN;
+import static com.hedera.services.utils.EntityIdUtils.isAlias;
+import static com.hedera.services.utils.EntityNum.MISSING_NUM;
+import static com.hedera.services.utils.EntityNum.fromAccountId;
+import static com.hedera.services.utils.EntityNum.fromContractId;
+import static com.hedera.services.utils.EntityNum.fromTokenId;
+import static com.hedera.services.utils.EntityNum.fromTopicId;
+
+public class StateChildrenSigMetadataLookup implements SigMetadataLookup {
```
Comment (qnswirlds) [inline hedera-node/src/main/java/com/hedera/services/sigs/metadata/StateChildrenSigMetadataLookup.java:60]: nit: this class could be made `final`

File: hedera-node/src/main/java/com/hedera/services/sigs/metadata/TokenMetaUtils.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.state.merkle.MerkleToken;
+import com.hedera.services.state.submerkle.FcCustomFee;
+
+public class TokenMetaUtils {
```
Comment (qnswirlds) [inline hedera-node/src/main/java/com/hedera/services/sigs/metadata/TokenMetaUtils.java:26]: nit: utils class should be `final`

File: hedera-node/src/main/java/com/hedera/services/sigs/metadata/TokenMetaUtils.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+	private TokenMetaUtils() {
+		throw new UnsupportedOperationException("Utility Class");
+	}
```
Comment (qnswirlds) [inline hedera-node/src/main/java/com/hedera/services/sigs/metadata/TokenMetaUtils.java:29]: Add this class to `UtilsConstructorTest`

File: hedera-node/src/test/java/com/hedera/services/sigs/metadata/lookups/HfsSigMetaLookupTest.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/src/test/java/com/hedera/services/sigs/metadata/lookups/HfsSigMetaLookupTest.java:5]: Removed intentionally, as `HfsSigMetaLookUp` will also be removed in future PRs after removing dependency in `SigRequirementsTest`

**Review by qnswirlds — APPROVED** [review]
Overall LGTM!

**Pull request merged**